platforms/jvm/build.gradle.kts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id("gradlebuild.gradle-platform")
+}
+
+dependencies {
```

Comment:
I don't really like how we need to list these projects again. 
They are already listed in `settings.gradle.kts`

I tried to do something like
```
subprojects.forEach {
    dependencies {
        subproject(it)
    }
}
```

However that does not work, since these are not true subprojects of this project. We determine subprojects based on path, so the above code would only work if we moved these to true subprojects (`:ear` -> `:jvm:ear`)

Reply:
`allprojects` is quite confusingly named. It is just `subprojects` + this project.

Reply:
Exactly.  At least to begin with.  Maybe later each platform could have a convention plugin that we could test for?

Reply:
The convention plugin option seems like it would require us to have a unique convention plugin per project. That seems a bit odd to me. 

IMO we should make decisions about what our subprojects are without needing to evaluate the other project -- that means using data that is declared in the settings file. AKA: project name (path), and root directory. 

Ideally I think these should be actual subprojects. Meaning `:plugins-java` -> `:jvm:plugins-java`. For now I think the file path comparison might be my preferred solution